scripts/updateCredentials.js

```diff
@@ -0,0 +1,143 @@
+ * Create a set of options for this script.
+ * The options are based on the command line parameters and a set of database constants.
+ * @param {Array} processArgv - The command line arguments.
+ * @return {Object} - The options.
+ */
+gpii.credentialsUpdater.initOptions = function (processArgv) {
+    var options = {};
+
+    options.couchDbUrl = processArgv[2] + "/gpii";
+    options.clientCredentialId = processArgv[3];
+    options.updateData = gpii.credentialsUpdater.loadUpdateData(processArgv[4]);
+
+    // Set up database specific options
+    options.parsedCouchDbUrl = url.parse(options.couchDbUrl);
+
+    fluid.log("COUHDB_URL: '" +
+        options.parsedCouchDbUrl.protocol + "//" +
+        options.parsedCouchDbUrl.hostname + ":" +
+        options.parsedCouchDbUrl.port +
+        options.parsedCouchDbUrl.pathname + "'"
+    );
+
+    return options;
+};
+
+/**
+ * Load the JSON update data from the provided filename.
+ * It tries to load the data from the specified file, if the file doesn't exist
+ * or doesn't contain any JSON data, the script will exit.
+ * @param {String} filename - The filename with the data to update.
+ * @return {Object} - The JSON data that will be used to update the document.
+ */
+gpii.credentialsUpdater.loadUpdateData = function (filename) {
+    if (!fs.existsSync(filename)) {
+        fluid.fail("The file '", filename, "' doesn't exist. Check that the file exists and try again");
+    }
+
+    try {
+        var rawData = fs.readFileSync(filename);
+        var updateData = JSON.parse(rawData);
+        return updateData;
+    } catch (err) {
+        fluid.fail("Couldn't load the update data. Check the content and try again");
+    }
+};
+
+/**
+ * Get the document that will be updated.
+ * @param {Object} options - The options to perform the request to CouchDB.
+ * @return {Promise} - A promise containing the retrieved document when resolved.
+ */
+gpii.credentialsUpdater.getDoc = function (options) {
+    var promise = fluid.promise();
+
+    var requestOptions = {
+        url: options.couchDbUrl + "/" + options.clientCredentialId,
+        json: true
+    };
+
+    request.get(requestOptions, function (error, response, body) {
+        var err = error || body.error;
+        if (err) {
+            promise.reject(err);
+        } else {
+            promise.resolve(body);
+        }
+    });
+
+    return promise;
+};
+
+/**
+ * Update the document.
+ * It updates the document with the given updateData.
+ * @param {Object} options - The options to perform the request to CouchDB.
+ * @param {Object} doc - The document that will be updated.
+ */
+gpii.credentialsUpdater.updateDoc = function (options, doc) {
+    var updatedDocData = fluid.extend(doc, options.updateData);
+
+    var requestOptions = {
+        url: options.couchDbUrl + "/" + options.clientCredentialId,
+        body: updatedDocData,
+        json: true
+    };
+
+    request.put(requestOptions, function (error /* response, body */) {
```

Comment:
It's not wise to hide the `body` value. The CouchDB error message could also be reported via it. For example, when a unmatched revision id is provided at a update, `body` value will receive: 
```
{
	error: "conflict",
	reason: "Document update conflict."
}
```
The error handling at line 120 should also take it into account.